src/modules/claim/claim.service.ts

```diff
@@ -352,4 +356,10 @@ export class ClaimService {
     });
     return claims.map(claim => claim.requester);
   }
+
+  static idOfClaim(claimReq: IClaimRequest) {
+    const { token, claimType: role, claimTypeVersion: version } = claimReq;
+    const { sub: subject } = jwt.decode(token);
+    return v5(JSON.stringify({ subject, role, version }), UUID_NAMESPACE);
```

Comment:
This is deterministic then? For the the same subject, role & version, it will create the same uuid?

Reply:
> This is deterministic then? For the the same subject, role & version, it will create the same uuid?

Yes, it should prevent create same enrollment request